Scripts/summaryStatistics/biophysics/growth_rate_analysis.py

```diff
@@ -0,0 +1,247 @@
+    
+def get_binned_statistics(df):
+    """
+    Calculate the mean and standard deviation of binned data
+    
+    @param  df          dataFrame with a column ['bin']; the column contains integers
+    @return mean, std   Group dataFrames: mean and standard deviation of measurements (float)
+    """
+    # Group each cell by bin
+    df_groupby_bin = df.groupby(df['bin'])
+    
+    # Calculate statistics for all other measurements
+    mean = df_groupby_bin.mean()
+    std = df_groupby_bin.std()
+    
+    return mean, std
+    
+def filter_by_cell_age(cells, min_age=2):
+    """
+    Create a dict with cells that have age >= min_age
+    
+    @param  cells               cellStates dict
+    @param  min_age             minimum age to include cell in the filtered list; recommended age is 2
+    @return filtered_cellstates cellStates dict with cells filtered by age
+    """
+    filtered_cellstates = {}
+    for i, cell in cells.items():
+        if cell.cellAge >= min_age:
+            filtered_cellstates[i] = cell
+    
+    return filtered_cellstates
+    
+def get_all_centroids(cells):
+    """
+    Obtain centroids for all cells
+    
+    @param  cells           cellStates dict
+    @return all_centroids   nparray (n_cells, 2) of cell centroids
+    """
+    # Initialize storage variables
+    n_cells = len(cells.keys())
+    all_centroids = np.zeros((n_cells, 2))
+    
+    # Obtain centroids for all cells
+    for i, cell in enumerate(cells.values()):
+        all_centroids[i][0] = cell.pos[0]
+        all_centroids[i][1] = cell.pos[1]
+        
+    return all_centroids
+    
+def get_colony_centroid(cells):
+    '''
+    Fit an ellipse around colony using the welzl function
+    
+    @param  cells            cellStates dict
+    @return centerX, centerY colony centroid coordinates [x0, y0]
+    '''
+    # Get cell centroids
+    cell_centroids = get_all_centroids(cells)
+    
+    # Required check for larger colonies
+    if len(cell_centroids) >= 1000:
+        sys.setrecursionlimit(10000) # necessary to run welzl for colonies with n_cells > 1000  
+    
+    # Fit ellipse     
+    center, major, minor, theta = welzl(cell_centroids)  
+    
+    centerX = center[0]
+    centerY = center[1]  
+    
+    return centerX, centerY   
+    
+def find_knee(x, y, curvature_type="convex", direction_type="increasing"):
+    """
+    Finds point of maximum curvature in x, y data
+    https://raghavan.usc.edu/papers/kneedle-simplex11.pdf
+    https://pypi.org/project/kneed
+    
+    @param  x               X-coordinates of data
+    @param  y               Y-coordinates of data
+    @param  curvature_type  (string) "convex" or "concave"  
+    @param  direction_type  (string) "increasing" or "decreasing"
+    @return knee            X-coordinate of the knee
+    """
+    
+    kneedle = KneeLocator(x, y, S=1.0, curve=curvature_type, direction=direction_type)
+    knee = kneedle.knee
+    
+    return knee
+    
+def fit_line(x, y):
```

Comment:
I believe "fit_line" is not a good name for this function since in statistical literature it's mentioned like "fit_straight_line" or/and "fit_linear_regression".